src/plasma/eviction_policy.cc

```diff
@@ -112,47 +78,33 @@ int64_t EvictionState_choose_objects_to_evict(
     int64_t num_bytes_required,
     int64_t *num_objects_to_evict,
     ObjectID **objects_to_evict) {
-  int64_t num_objects = 0;
-  int64_t num_bytes = 0;
-  /* Figure out how many objects need to be evicted in order to recover a
-   * sufficient number of bytes. */
-  ReleasedObject *element, *temp;
-  DL_FOREACH_SAFE(eviction_state->released_objects, element, temp) {
-    if (num_bytes >= num_bytes_required) {
-      break;
-    }
-    /* Find the object table entry for this object. */
-    ObjectTableEntry *entry = plasma_store_info->objects[element->object_id];
-    /* Update the cumulative bytes and the number of objects so far. */
-    num_bytes += (entry->info.data_size + entry->info.metadata_size);
-    num_objects += 1;
+  std::vector<ObjectID> objs_to_evict;
+  int64_t bytes_evicted = eviction_state->cache.choose_objects_to_evict(
+      num_bytes_required, objs_to_evict);
+  /* Update the LRU cache. */
+  for (auto &object_id : objs_to_evict) {
+    eviction_state->cache.remove(object_id);
   }
   /* Construct the return values. */
-  *num_objects_to_evict = num_objects;
-  if (num_objects == 0) {
+  *num_objects_to_evict = objs_to_evict.size();
+  if (objs_to_evict.size() == 0) {
     *objects_to_evict = NULL;
   } else {
-    *objects_to_evict = (ObjectID *) malloc(num_objects * sizeof(ObjectID));
-    int counter = 0;
-    DL_FOREACH_SAFE(eviction_state->released_objects, element, temp) {
-      if (counter == num_objects) {
-        break;
-      }
-      (*objects_to_evict)[counter] = element->object_id;
-      /* Update the LRU cache. */
-      remove_object_from_lru_cache(eviction_state, element->object_id);
-      counter += 1;
-    }
+    int64_t result_size = objs_to_evict.size() * sizeof(ObjectID);
+    *objects_to_evict = (ObjectID *) malloc(result_size);
+    memcpy(*objects_to_evict, objs_to_evict.data(), result_size);
```

Comment:
This is valid use of vectors, but I'll remove it in the next big PR, see https://github.com/ray-project/ray/pull/535